### PR TITLE
Add session.Close & rename GetPayChsInfo in grpc

### LINKED ISSUE #111: Implement Session Close
#### Motivation / Context

To implement the session close function with proper handling of channels in different phases of their lifecycle.

#### Description

Currently, [session close](https://github.com/hyperledger-labs/perun-node/blob/c2451fc5f30d065f02b6c5f66e61ab34f6f37846/session/session.go#L514) is a No-op. Implement a session close mechanism with the following behavior:
1. When `force=true`, channels in all phases are locked, `channelClient` is closed and return a list of channels that are in open phase.
2. When `force=false`,
   - If there are channels in open phase, return an error with the list of channels in open phase.
   - If there are not channels in open phase, lock all the closed channels, close the `channelClient`.

#### Testing

1. Ensure channels in all phases of their lifecycle are handled properly.

#### Thread Safety

1. The `session.Close` should ensure the ongoing operations in any channel belonging to this session is completed.
2. The `session.Close` should ensure the ongoing operations in `statemachine` of `go-perun/channel` that was not initiated by `perun-node` is also completed. Currently only one such events exist: Watcher handling a registered event.

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

- Implement Session.Close with force option. Add tests in session,
app/payment and api/grpc packages. Update Error message definition in
proto buff, to also include a list of openChsInfo.

- Rename GetPayChs API in grpc package to GetPayChsInfo.

- Change concurrency pattern in session/role_integ_test.go to call
wg.Wait() in defer statement instead of end of the code section. This is
because, in previous pattern (calling at end), Wait() is never called if
any of the "require" assertions fail, ending the test before the
go-routine can finish.

- Also unexport the following conversion functions as they are not used
outside of the respective packages: toPayChInfo, toPayChsInfo in
app/payment and toGrpcXXXX functions in api/grpc package.

##### Category

Implementation Task

##### Relevant issue

Closes #111.

#### Testing

Added tests (as sub tests) to `Test_Integ_Role` test in `session` package and `api/grpc` package.
Also added tests in payment package to cover session close.

##### Steps to run the tests

1. Start the ganache cli-node:   `ganache-cli -b 1 --account="0x1fedd636dbc7e8d41a0622a2040b86fea8842cef9d4aa4c582aad00465b7acff,100000000000000000000" \
--account="0xb0309c60b4622d3071fad3e16c2ce4d0b1e7758316c187754f4dd0cfb44ceb33,100000000000000000000"`
2. Run all tests, including integration tests: ` go test -tags=integration -count=1 -cover  ./...`

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

**Comment by manoranjith** [comment]
Rebased onto develop.

**Comment by manoranjith** [comment]
Rebased onto develop after another PR was merged.

File: app/payment/session.go [hunk 1]
```diff
@@ -92,5 +92,11 @@ func UnsubPayChProposals(s perun.SessionAPI) error {
 // RespondPayChProposal sends the response to a payment channel proposal notification.
 func RespondPayChProposal(pctx context.Context, s perun.SessionAPI, proposalID string, accept bool) (PayChInfo, error) {
 	chInfo, err := s.RespondChProposal(pctx, proposalID, accept)
-	return ToPayChInfo(chInfo), err
+	return toPayChInfo(chInfo), err
+}
+
+// CloseSession sessions closes the current session.
```
Comment (anagha-ks) [inline app/payment/session.go]:
```suggestion
// CloseSession closes the current session.
```

File: api/grpc/server_integ_test.go [hunk 2]
```diff
@@ -310,51 +355,79 @@ func OpenPayCh(t *testing.T, sessionID string, parts, bal []string, wantErr bool
...
 	return msg.MsgSuccess.OpenedPayChInfo.ChID
 }
 
-func SubPayChProposal(t *testing.T, sessionID string) pb.Payment_API_SubPayChProposalsClient {
+func SubPayChProposal(t *testing.T, sessionID string, wantErr bool) pb.Payment_API_SubPayChProposalsClient {
 	subReq := pb.SubPayChProposalsReq{
 		SessionID: sessionID,
 	}
 	subClient, err := client.SubPayChProposals(ctx, &subReq)
+	// if wantErr {
```
Comment (anagha-ks) [inline api/grpc/server_integ_test.go]: Delete the `if` condition if not required.

File: api/grpc/server_integ_test.go [hunk 3]
```diff
@@ -132,8 +132,9 @@ func Test_Integ_Role(t *testing.T) {
 	var alicePeer, bobPeer *pb.Peer
 	var chID string
 	prng := rand.New(rand.NewSource(1729))
-	aliceCfgFile := sessiontest.NewConfigFile(t, sessiontest.NewConfig(t, prng))
-	bobCfgFile := sessiontest.NewConfigFile(t, sessiontest.NewConfig(t, prng))
+	aliceCfg, bobCfg := sessiontest.NewConfig(t, prng), sessiontest.NewConfig(t, prng)
```
Comment (anagha-ks) [inline api/grpc/server_integ_test.go]: Why this is changed so?

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: api/grpc/server_integ_test.go (hunk 3 above)
Comment (manoranjith) [reply api/grpc/server_integ_test.go]: I was changing somethings else and forgotten to undo it I guess. Reverting to the older state.
```suggestion
	aliceCfgFile := sessiontest.NewConfigFile(t, sessiontest.NewConfig(t, prng))
	bobCfgFile := sessiontest.NewConfigFile(t, sessiontest.NewConfig(t, prng))
```

File: api/grpc/server_integ_test.go (hunk 2 above)
Comment (manoranjith) [reply api/grpc/server_integ_test.go]: Done

**Comment by manoranjith** [comment]
Squashed the review commits, as they were failing DCO check.

**Review by anagha-ks — APPROVED** [review]

**Pull request merged**